numpy/core/src/umath/loops.c.src

```diff
@@ -1077,11 +1078,26 @@ NPY_NO_EXPORT void
     }
 }
 
+/**begin repeat1
+ * #kind = gcd, lcm#
+ **/
+NPY_NO_EXPORT void
+@TYPE@_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    BINARY_LOOP {
+        const @type@ in1 = *(@type@ *)ip1;
+        const @type@ in2 = *(@type@ *)ip2;
+        *((@type@ *)op1) = npy_@kind@@c@(in1, in2);
+    }
+}
+/**end repeat1**/
+
 /**end repeat**/
 
 /**begin repeat
  * #TYPE = UBYTE, USHORT, UINT, ULONG, ULONGLONG#
  * #type = npy_ubyte, npy_ushort, npy_uint, npy_ulong, npy_ulonglong#
+ * #c    = u,u,u,ul,ull#
```

Comment:
Possibly. Although I didn't see much point implementing `B`, `H`, and `I` separately, because arithmetic always promotes to `int` in C anyway, doesn't it?

Can you link me to a file that uses these standard codes?